### ISSUE TITLE
azurerm_automation_dsc_nodeconfiguration_test location warnings

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#784]
`INFRACOST_LOG_LEVEL=info go test -v internal/providers/terraform/azure/automation_dsc_nodeconfiguration_test.go` shows this, which is odd as underneath, that resource calls `NewAzureRMAutomationDscConfiguration`, which does not have this issue. This might be related to how the Terraform plan JSON represents this the node resource. I've included the 2 plan JSON files in this ticket so they can be compared.

```
time="2021-06-07T22:03:01+01:00" level=warning msg="Using eastus for resource azurerm_automation_dsc_nodeconfiguration.fiveNodes as its 'location' property could not be found."
time="2021-06-07T22:03:01+01:00" level=warning msg="Using eastus for resource azurerm_automation_dsc_nodeconfiguration.withoutUsage as its 'location' property could not be found."
time="2021-06-07T22:03:01+01:00" level=warning msg="Using eastus for resource azurerm_automation_dsc_nodeconfiguration.zeroNodes as its 'location' property could not be found."
```

automation_dsc_configuration_test:
```
{"format_version":"0.1","terraform_version":"0.15.4","planned_values":{"root_module":{"resources":[{"address":"azurerm_automation_account.example","mode":"managed","type":"azurerm_automation_account","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"location":"eastus","name":"account1","resource_group_name":"example-resources","sku_name":"Basic","tags":null,"timeouts":null}},{"address":"azurerm_automation_dsc_configuration.fiveNodes","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"fiveNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"test","resource_group_name":"example-resources","tags":null,"timeouts":null}},{"address":"azurerm_automation_dsc_configuration.without_usage","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"without_usage","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"test","resource_group_name":"example-resources","tags":null,"timeouts":null}},{"address":"azurerm_automation_dsc_configuration.zeroNodes","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"zeroNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"test","resource_group_name":"example-resources","tags":null,"timeouts":null}},{"address":"azurerm_resource_group.example","mode":"managed","type":"azurerm_resource_group","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"location":"eastus","name":"example-resources","tags":null,"timeouts":null}}]}},"resource_changes":[{"address":"azurerm_automation_account.example","mode":"managed","type":"azurerm_automation_account","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"location":"eastus","name":"account1","resource_group_name":"example-resources","sku_name":"Basic","tags":null,"timeouts":null},"after_unknown":{"dsc_primary_access_key":true,"dsc_secondary_access_key":true,"dsc_server_endpoint":true,"id":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_configuration.fiveNodes","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"fiveNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"test","resource_group_name":"example-resources","tags":null,"timeouts":null},"after_unknown":{"id":true,"state":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_configuration.without_usage","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"without_usage","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"test","resource_group_name":"example-resources","tags":null,"timeouts":null},"after_unknown":{"id":true,"state":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_configuration.zeroNodes","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"zeroNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"test","resource_group_name":"example-resources","tags":null,"timeouts":null},"after_unknown":{"id":true,"state":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_resource_group.example","mode":"managed","type":"azurerm_resource_group","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"location":"eastus","name":"example-resources","tags":null,"timeouts":null},"after_unknown":{"id":true},"before_sensitive":false,"after_sensitive":{}}}],"configuration":{"provider_config":{"azurerm":{"name":"azurerm","expressions":{"features":[{}],"skip_provider_registration":{"constant_value":true}}}},"root_module":{"resources":[{"address":"azurerm_automation_account.example","mode":"managed","type":"azurerm_automation_account","name":"example","provider_config_key":"azurerm","expressions":{"location":{"references":["azurerm_resource_group.example"]},"name":{"constant_value":"account1"},"resource_group_name":{"references":["azurerm_resource_group.example"]},"sku_name":{"constant_value":"Basic"}},"schema_version":0},{"address":"azurerm_automation_dsc_configuration.fiveNodes","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"fiveNodes","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"location":{"references":["azurerm_resource_group.example"]},"name":{"constant_value":"test"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_automation_dsc_configuration.without_usage","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"without_usage","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"location":{"references":["azurerm_resource_group.example"]},"name":{"constant_value":"test"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_automation_dsc_configuration.zeroNodes","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"zeroNodes","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"location":{"references":["azurerm_resource_group.example"]},"name":{"constant_value":"test"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_resource_group.example","mode":"managed","type":"azurerm_resource_group","name":"example","provider_config_key":"azurerm","expressions":{"location":{"constant_value":"eastus"},"name":{"constant_value":"example-resources"}},"schema_version":0}]}}}
```

automation_dsc_nodeconfiguration_test:
```
{"format_version":"0.1","terraform_version":"0.15.4","planned_values":{"root_module":{"resources":[{"address":"azurerm_automation_account.example","mode":"managed","type":"azurerm_automation_account","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"location":"eastus","name":"account1","resource_group_name":"example-resources","sku_name":"Basic","tags":null,"timeouts":null}},{"address":"azurerm_automation_dsc_configuration.example","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"example","resource_group_name":"example-resources","tags":null,"timeouts":null}},{"address":"azurerm_automation_dsc_nodeconfiguration.fiveNodes","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"fiveNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","name":"fiveNodes","resource_group_name":"example-resources","timeouts":null}},{"address":"azurerm_automation_dsc_nodeconfiguration.withoutUsage","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"withoutUsage","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","name":"withoutUsage","resource_group_name":"example-resources","timeouts":null}},{"address":"azurerm_automation_dsc_nodeconfiguration.zeroNodes","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"zeroNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"automation_account_name":"account1","content_embedded":"configuration test {}","name":"zeroNodes","resource_group_name":"example-resources","timeouts":null}},{"address":"azurerm_resource_group.example","mode":"managed","type":"azurerm_resource_group","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","schema_version":0,"values":{"location":"eastus","name":"example-resources","tags":null,"timeouts":null}}]}},"resource_changes":[{"address":"azurerm_automation_account.example","mode":"managed","type":"azurerm_automation_account","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"location":"eastus","name":"account1","resource_group_name":"example-resources","sku_name":"Basic","tags":null,"timeouts":null},"after_unknown":{"dsc_primary_access_key":true,"dsc_secondary_access_key":true,"dsc_server_endpoint":true,"id":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_configuration.example","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","description":null,"location":"eastus","log_verbose":false,"name":"example","resource_group_name":"example-resources","tags":null,"timeouts":null},"after_unknown":{"id":true,"state":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_nodeconfiguration.fiveNodes","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"fiveNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","name":"fiveNodes","resource_group_name":"example-resources","timeouts":null},"after_unknown":{"configuration_name":true,"id":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_nodeconfiguration.withoutUsage","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"withoutUsage","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","name":"withoutUsage","resource_group_name":"example-resources","timeouts":null},"after_unknown":{"configuration_name":true,"id":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_automation_dsc_nodeconfiguration.zeroNodes","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"zeroNodes","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"automation_account_name":"account1","content_embedded":"configuration test {}","name":"zeroNodes","resource_group_name":"example-resources","timeouts":null},"after_unknown":{"configuration_name":true,"id":true},"before_sensitive":false,"after_sensitive":{}}},{"address":"azurerm_resource_group.example","mode":"managed","type":"azurerm_resource_group","name":"example","provider_name":"registry.terraform.io/hashicorp/azurerm","change":{"actions":["create"],"before":null,"after":{"location":"eastus","name":"example-resources","tags":null,"timeouts":null},"after_unknown":{"id":true},"before_sensitive":false,"after_sensitive":{}}}],"configuration":{"provider_config":{"azurerm":{"name":"azurerm","expressions":{"features":[{}],"skip_provider_registration":{"constant_value":true}}}},"root_module":{"resources":[{"address":"azurerm_automation_account.example","mode":"managed","type":"azurerm_automation_account","name":"example","provider_config_key":"azurerm","expressions":{"location":{"references":["azurerm_resource_group.example"]},"name":{"constant_value":"account1"},"resource_group_name":{"references":["azurerm_resource_group.example"]},"sku_name":{"constant_value":"Basic"}},"schema_version":0},{"address":"azurerm_automation_dsc_configuration.example","mode":"managed","type":"azurerm_automation_dsc_configuration","name":"example","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"location":{"references":["azurerm_resource_group.example"]},"name":{"constant_value":"example"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_automation_dsc_nodeconfiguration.fiveNodes","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"fiveNodes","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"name":{"constant_value":"fiveNodes"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_automation_dsc_nodeconfiguration.withoutUsage","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"withoutUsage","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"name":{"constant_value":"withoutUsage"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_automation_dsc_nodeconfiguration.zeroNodes","mode":"managed","type":"azurerm_automation_dsc_nodeconfiguration","name":"zeroNodes","provider_config_key":"azurerm","expressions":{"automation_account_name":{"references":["azurerm_automation_account.example"]},"content_embedded":{"constant_value":"configuration test {}"},"name":{"constant_value":"zeroNodes"},"resource_group_name":{"references":["azurerm_resource_group.example"]}},"schema_version":0},{"address":"azurerm_resource_group.example","mode":"managed","type":"azurerm_resource_group","name":"example","provider_config_key":"azurerm","expressions":{"location":{"constant_value":"eastus"},"name":{"constant_value":"example-resources"}},"schema_version":0}]}}}
```

**Comment by alikhajeh1** [comment]
@NeverovDS Alistair found the cause... we need to add `ReferenceAttributes` to **all** Azure resources, otherwise the `lookupRegion` method will not have access to the resource_group_name. Can you possibly do that?

```
func GetAzureRMAppServiceCustomHostnameBindingRegistryItem() *schema.RegistryItem {
	return &schema.RegistryItem{
		Name:  "azurerm_app_service_custom_hostname_binding",
		RFunc: NewAzureRMAppServiceCustomHostnameBinding,
		ReferenceAttributes: []string{"resource_group_name"}, <<<<< this is needed, if they already have `ReferenceAttributes` then we should add `resource_group_name` to the list
	}
}
```

**Comment by alikhajeh1** [comment]
@NeverovDS is this issue resolved? if so, plz feel free to close it

**Comment by alikhajeh1** [comment]
Released in v0.9.1
